namerd/storage/zk/src/main/scala/com/twitter/finagle/serverset2/buoyant/ZkSession.scala

```diff
@@ -0,0 +1,186 @@
+// modified from com.twitter.finagle.serverset2.ZkSession | (c) 2015 Twitter, Inc. | http://www.apache.org/licenses/LICENSE-2.0 */
+package com.twitter.finagle.serverset2.buoyant
+
+import com.twitter.concurrent.AsyncSemaphore
+import com.twitter.finagle.serverset2.RetryStream
+import com.twitter.finagle.serverset2.client._
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.logging.Logger
+import com.twitter.util._
+
+/**
+  * ZkSession wraps a ZooKeeper client which automatically reconnects upon session expiry.
+  * Cribbed from https://github.com/twitter/finagle/blob/develop/finagle-serversets/src/main/scala/com/twitter/finagle/serverset2/ZkSession.scala
+  */
```

Comment:
It might be good to add some comments noting which portions of the code are fully copied from the Finagle implementation and which are specific to this one.